lib/segment/src/index/hnsw_index/graph_links.rs

```diff
@@ -217,42 +257,87 @@ impl GraphLinksConverter {
 
         {
             let level_offsets_range = header.get_level_offsets_range();
-            let level_offsets_byte_slice = &mut mmap[level_offsets_range];
+            let level_offsets_byte_slice = &mut bytes_data[level_offsets_range];
             let level_offsets_slice: &mut [u64] = transmute_from_u8_mut(level_offsets_byte_slice);
             level_offsets_slice.copy_from_slice(&level_offsets);
         }
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut bytes = vec![0; self.data_size() as usize];
+        self.serialize_to(&mut bytes);
+        bytes
+    }
+
+    pub fn save_as(&mut self, path: &Path) -> OperationResult<()> {
+        self.path = Some(path.to_path_buf());
+        let tmp_path = path.with_extension("tmp");
+        {
+            let file = OpenOptions::new()
+                .read(true)
+                .write(true)
+                .create(true)
+                .open(tmp_path.as_path())?;
+
+            file.set_len(self.data_size())?;
+            let m = unsafe { MmapMut::map_mut(&file) };
+            let mut mmap = m?;
+
+            self.serialize_to(&mut mmap);
+
+            mmap.flush()?;
+        }
+        std::fs::rename(tmp_path, path)?;
```

Comment:
Do not override the original data right away, conversion might fail and we don't want to loose the data.